Assets/Scripts/Calendar UI.cs

```diff
@@ -23,89 +25,136 @@ public class CalendarUI : MonoBehaviour
     [SerializeField] private TextMeshProUGUI yearText;
 
 
+
     private DateTime currentDate;
+    private DateTime defaultDate;
     private DateTime firstDayOfMonth;
+
     private int daysInMonth;
     private int numberOfBlanksBefore;
     private int numberOfBlanksAfter;
+
     private void Start()
     {
-
+        EventManager.instance.onForwardClick += HandleForwardClick;
+        EventManager.instance.onBackwardClick += HandleBackwardClick;
         currentDate = DateTime.Today;
+        defaultDate = currentDate;
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+
 
-        monthText.text = currentDate.ToString("MMMM");
 
-        yearText.text = currentDate.ToString("yyyy");
-        firstDayOfMonth = new DateTime(currentDate.Year, currentDate.Month, 1);
-        daysInMonth = DateTime.DaysInMonth(currentDate.Year, currentDate.Month);
         numberOfBlanksBefore = GetBlanksBefore();
         numberOfBlanksAfter = GetBlanksAfter();
+        PopulateCalendarGrid();
         PopulateWeekDaysGrid();
+
+    }
+
+    private void OnDestroy()
+    {
+        EventManager.instance.onForwardClick -= HandleForwardClick;
+        EventManager.instance.onBackwardClick -= HandleBackwardClick;
+    }
+
+
+    private void HandleForwardClick()
+    {
+        calendarGridPopulator.ClearCalendarGrid();
+
+
+        defaultDate = defaultDate.AddMonths(1);
+        defaultDate = new DateTime(defaultDate.Year, defaultDate.Month, 1);
+
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+        numberOfBlanksBefore = GetBlanksBefore();
+        numberOfBlanksAfter = GetBlanksAfter();
+
         PopulateCalendarGrid();
+
     }
+    private void HandleBackwardClick()
+    {
+        calendarGridPopulator.ClearCalendarGrid();
+        defaultDate = defaultDate.AddMonths(-1);
 
 
+        monthText.text = defaultDate.ToString("MMMM");
+        yearText.text = defaultDate.ToString("yyyy");
+        numberOfBlanksBefore = GetBlanksBefore();
+        numberOfBlanksAfter = GetBlanksAfter();
+        PopulateCalendarGrid();
+    }
 
     public void PopulateCalendarGrid()
     {
 
         int totalDays = numberOfBlanksBefore + daysInMonth + numberOfBlanksAfter;
-        List<GameObject> populationList = GridPopulator.instance.PopulateTheGrid(totalDays, calendarGrid, dayPrefab);
+        List<Cell> populationList = calendarGridPopulator.PopulateTheGrid(totalDays);
+        int day = 1;
         for (int i = 0; i <= populationList.Count - 1; i++)
         {
 
 
             if (i < numberOfBlanksBefore || i >= numberOfBlanksBefore + daysInMonth)
             {
-                populationList[i].GetComponent<Image>().color = blankColor;
-                GridPopulator.instance.SetTextValue("", populationList[i]);
+                populationList[i].SetImageColor(blankColor);
```

Comment:
This warrants an extra ticket as its a bit more work, and this is currently functional.

We are probably going to want to pass in a DateTime to these cells in the future, and have them configure themselves & their colour based on that datetime, rather than the parents giving them all the information they need. I just want to bring it up here that this will cause problems down the line, and to have a conversation with me prior to adding more functionality for the cell abstract class.

Reply:
Resolved